decode.go

```diff
@@ -63,6 +67,10 @@ type PacketBuilder interface {
 	// data will be dumped to stderr so you can create a test.  This should never
 	// be called from a production decoder.
 	DumpPacketData()
+	// DecodeApplicationLayers returns true if the packet was
+	// created with a decode option to enable application-level
+	// decoding (currently only checked by the TCP decoder).
+	DecodeApplicationLayers() bool
```

Comment:
Let's instead do:

```go
DecodeOptions() *DecodeOptions
```

That way, as other options are added to DecodeOptions in the future, additional functions aren't necessary here.